core/plannernotes.c

```diff
@@ -265,28 +293,54 @@ void add_plan_to_notes(struct diveplan *diveplan, struct dive *dive, bool show_d
 					len += snprintf(buffer + len, sz_buffer - len, "<td style='padding-left: 10px; float: right;'>%s</td>", temp);
 				}
 
+				if (istrimix) {
+					runtime = (dp->time + 30) / 60;
+					deltaN2 = ((1000 - newgasmix.he.permille - newgasmix.o2.permille) - (1000 - lastprintgasmix.he.permille - lastprintgasmix.o2.permille)) / 10.0;
+					deltaHe = (newgasmix.he.permille - lastprintgasmix.he.permille) / 10.0;
+					maxdeltaN2 = 0.02 * (lastprintgasmix.he.permille-newgasmix.he.permille);
+					if (deltaN2 > maxdeltaN2) icdwarning = true;
+					deltaPN2 = depth_to_bar(dp->depth.mm, dive) * deltaN2 / 100;
+					deltaPHe = depth_to_bar(dp->depth.mm, dive) * deltaHe / 100;
+					maxdeltaPN2 = depth_to_bar(dp->depth.mm, dive) * maxdeltaN2 / 100;
+				}
+
 				/* Normally a gas change is displayed on the stopping segment, so only display a gas change at the end of
 				 * an ascent segment if it is not followed by a stop
 				 */
+
 				if ((isascent || dp->entered) && gaschange_after && dp->next && nextdp && (dp->depth.mm != nextdp->depth.mm || nextdp->entered)) {
 					if (dp->setpoint) {
 						snprintf(temp, sz_temp, translate("gettextFromC", "(SP = %.1fbar)"), (double) nextdp->setpoint / 1000.0);
 						len += snprintf(buffer + len, sz_buffer - len, "<td style='padding-left: 10px; color: red; float: left;'><b>%s %s</b></td>", gasname(&newgasmix),
 									temp);
 					} else {
-							len += snprintf(buffer + len, sz_buffer - len, "<td style='padding-left: 10px; color: red; float: left;'><b>%s</b></td>", gasname(&newgasmix));
+						len += snprintf(buffer + len, sz_buffer - len, "<td style='padding-left: 10px; color: red; float: left;'><b>%s</b></td>", gasname(&newgasmix));
+						if (isascent && (lastprintgasmix.he.permille > 0)) {		// On ascent, save ICD info if previous cylinder had helium
+							icdlen += snprintf(icdbuffer + icdlen, sz_icdbuf - icdlen, translate("gettextFromC", 
+							"<tr><td rowspan='2' style= 'vertical-align:middle;'>%3dmin</td><td rowspan=2 style= 'vertical-align:middle;'>%s&#10137;%s</td><td style='padding-left: 10px;'>%+5.2f %</td><td style= 'padding-left: 15px; color:%s;'>%+5.2f %</td><td style='padding-left: 15px;'>%+5.2f %</td></tr> <tr><td style='padding-left: 10px;'>%+5.2f bar</td><td style='padding-left: 15px; color:%s;'>%+5.2f bar</td><td style='padding-left: 15px;'>%+5.2f bar</td></tr>"),
```

Comment:
Strings with so much markup tend to get mangled in translation. In the end, it's only the "bar" that needs translation. You could leave the main string untranslated and get the bar as a %s string and have only that translated.